docs/root/configuration/http/http_filters/bandwidth_limit_filter.rst

```diff
@@ -0,0 +1,100 @@
+.. _config_http_filters_bandwidth_limit:
+
+HTTP Bandwidth limit
+====================
+
+* Bandwidth limiting :ref:`architecture overview <arch_overview_bandwidth_limit>`
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.bandwidth_limit.v3alpha.BandwidthLimit>`
+* This filter should be configured with the name *envoy.filters.http.bandwidth_limit*.
+
+The HTTP Bandwidth limit filter limits the size of data flow to the max bandwidth set in the limit_kbps
+when the request's route or virtual host has a per filter
+:ref:`bandwidth limit configuration <envoy_v3_api_msg_extensions.filters.http.bandwidth_limit.v3alpha.BandwidthLimit>`.
+
+If the bandwidth limit has been exhausted the filter stops further transfer until more bandwidth gets allocated
+according to the fill_interval(default is 50 milliseconds). If the connection buffer fills up with accumulated
+data then the source of data will be readDisabled as described in the :ref:`flow control doc<faq_flow_control>`.
+
+.. note::
+  The token bucket is shared across all workers, thus the limits are applied per Envoy process.
+
+Example configuration
+---------------------
+
+Example filter configuration for a globally set bandwidth limiter (e.g.: all vhosts/routes share the same token bucket):
+
+.. code-block:: yaml
+
+  name: envoy.filters.http.bandwidth_limit
+  typed_config:
+    "@type": type.googleapis.com/envoy.extensions.filters.http.bandwidth_limit.v3alpha.BandwidthLimit
+    stat_prefix: http_bandwidth_limiter
+    enable_mode: DecodeAndEncode
+    limit_kbps: 100
+    fill_interval: 0.1s
+
+Example filter configuration for a globally disabled bandwidth limiter but enabled for a specific route:
+
+.. code-block:: yaml
+
+  name: envoy.filters.http.bandwidth_limit
+  typed_config:
+    "@type": type.googleapis.com/envoy.extensions.filters.http.bandwidth_limit.v3alpha.BandwidthLimit
+    stat_prefix: http_bandwidth_limiter
+
+
+The route specific configuration:
+
+.. code-block:: yaml
+
+  route_config:
```

Comment:
it would be preferable to use a `literalinclude` with a full envoy config and only these lines showing - it just means that this config will be validated and tested